src/Areas/Storage/Services/StorageService.cs

```diff
@@ -321,4 +323,55 @@ private async Task<BlobServiceClient> CreateBlobServiceClient(string accountName
         }
         return new BlobServiceClient(new Uri(uri), await GetCredential(tenant), options);
     }
+
+    private async Task<DataLakeServiceClient> CreateDataLakeServiceClient(string accountName, string? tenant = null, RetryPolicyOptions? retryPolicy = null)
+    {
+        var uri = $"https://{accountName}.dfs.core.windows.net";
+        var options = AddDefaultPolicies(new DataLakeClientOptions());
+
+        if (retryPolicy != null)
+        {
+            options.Retry.Delay = TimeSpan.FromSeconds(retryPolicy.DelaySeconds);
+            options.Retry.MaxDelay = TimeSpan.FromSeconds(retryPolicy.MaxDelaySeconds);
+            options.Retry.MaxRetries = retryPolicy.MaxRetries;
+            options.Retry.Mode = retryPolicy.Mode;
+            options.Retry.NetworkTimeout = TimeSpan.FromSeconds(retryPolicy.NetworkTimeoutSeconds);
+        }
```

Comment:
We could benefit by adding this retry policy assignment to a shared library or base class so other services can use it too.